tools/souper-interpret.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright 2014 The Souper Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "llvm/Support/MemoryBuffer.h"
+#include "souper/Infer/Interpreter.h"
+#include "souper/Parser/Parser.h"
+#include "souper/Tool/GetSolverFromArgs.h"
+
+using namespace llvm;
+using namespace souper;
+
+static cl::list<std::string>
+InputValueStrings("input-values", cl::desc("<input values>"),
+                  cl::CommaSeparated);
+
+static cl::opt<std::string>
+InputFilename(cl::Positional, cl::desc("<input souper optimization>"),
+              cl::init("-"));
+
+static cl::opt<unsigned> DebugLevel("souper-debug-level", cl::init(1),
+     cl::desc("Control the verbose level of debug output (default=1). "
+     "The larger the number is, the more fine-grained debug "
+     "information will be printed."));
+
+static bool parseInput(const std::string &S, std::string &Name,
+                       std::string &Val) {
+  if (S[0] != '%')
+    return false;
+  int Pos = 1;
+  while (Pos < S.size() &&
+         (S[Pos] >= '0' && S[Pos] <= '9' ||
```

Comment:
More parentheses here and the next two lines.
CI reports a build failure otherwise, coming from a warning turned into an error.

Reply:
argh!! thanks